attestation-service/attestation-service/src/lib.rs

```diff
@@ -72,44 +122,54 @@ impl AttestationService {
             .map_err(|e| anyhow!("Cannot Set Policy: {:?}", e))
     }
 
-    fn accumulate_hash(materials: &[Vec<u8>]) -> Option<Vec<u8>> {
-        if materials.is_empty() {
-            return None;
-        }
-        let mut hasher = Sha384::new();
-        materials.iter().for_each(|m| hasher.update(m));
-        Some(hasher.finalize().to_vec())
-    }
-
     /// Evaluate Attestation Evidence.
-    /// Issue an attestation results token which contain TCB status and TEE public key.
-    pub async fn evaluate(&self, tee: Tee, nonce: &str, attestation: &str) -> Result<String> {
-        let attestation = serde_json::from_str::<Attestation>(attestation)
-            .context("Failed to deserialize Attestation")?;
+    /// Issue an attestation results token which contain TCB status and TEE public key. Input parameters:
+    /// - `evidence`: TEE evidence bytes. This might not be the raw hardware evidence bytes. Definitions
+    /// are in `verifier` crate.
+    /// - `tee`: concrete TEE type
+    /// - `runtime_data`: These data field will be used to check against the counterpart inside the evidence.
+    /// The concrete way of checking is decide by the enum type. If this parameter is set `None`, the comparation
+    /// will not be performed.
+    /// - `init_data`: These data field will be used to check against the counterpart inside the evidence.
+    /// The concrete way of checking is decide by the enum type. If this parameter is set `None`, the comparation
+    /// will not be performed.
+    /// - `hash_algorithm`: The hash algorithm that is used to calculate the digest of `runtime_data` and
+    /// `init_data`.
+    /// - `policy_ids`: The policy ids that used to check this evidence. Any check fails against a policy will
+    /// not cause this function to return error. The result check against every policy will be included inside
+    /// the finally Token returned by CoCo-AS.
+    /// - `customized_claims`: user defined JSON Map object that will be included in the result token.
+    /// AS will not do any extra check upon this parameter.
```

Comment:
Looks like this field as removed? We should update the comments.

Reply:
Thanks for this. Let me fix